openedx/core/djangoapps/content_libraries/models.py

```diff
@@ -227,6 +229,60 @@ def __str__(self):
         return f"ContentLibraryPermission ({self.access_level} for {who})"
 
 
+class ContentLibraryMigration(models.Model):
+    """
+    Record of a legacy (v1) content library that has been migrated into a new (v2) content library.
+    """
+    source_key = LearningContextKeyField(unique=True, max_length=255)
+    target = models.ForeignKey(ContentLibrary, on_delete=models.CASCADE)
+    target_collection = models.ForeignKey(Collection, on_delete=models.SET_NULL, null=True)
```

Comment:
This model encodes some very reasonable lifecycle behavior that is probably worth explicitly calling out in the comments, (e.g. "deleting the v2 library deletes any record of the migration", "collection is nullable because you can migrate and then delete the target collection, without actually deleting the components in that collection").